### ISSUE TITLE
Module 5: ONNX Export & Optimization (1 Pomodoro)

### ISSUE THREAD

**Issue by gregory1506** [gregory1506/arc-pytorch-prep#5]
## Module 5: ONNX Export & Optimization
**Duration**: 1 Pomodoro (25 minutes)  
**Status**: 🔄 Not Started  
**Priority**: Medium
**Depends on**: #3

### Learning Objectives
By the end of this module, students will be able to:
1. Export trained PyTorch models to ONNX format
2. Verify ONNX model produces same outputs as PyTorch
3. Understand benefits of ONNX for deployment
4. Run inference with ONNX Runtime
5. Compare performance (speed) of PyTorch vs ONNX

### Full Pomodoro: Teaching + Implementation + Testing
- [ ] **Teaching (8 min)**: Model deployment preparation
- [ ] **Code (10 min)**: Fill in src/optimization/optimize.py
  - TODO 5.1: Load trained checkpoint
  - TODO 5.2: Export to ONNX format
  - TODO 5.3: Verify ONNX model with onnxruntime
  - TODO 5.4: Benchmark inference speed
- [ ] **Test (5 min)**: Run pytest tests/test_module_05_onnx.py -v
- [ ] **Benchmark (2 min)**: Compare PyTorch vs ONNX inference time

### Files to Edit
- src/optimization/optimize.py

### Test Command
pytest tests/test_module_05_onnx.py -v

### Teaching Notes
See TEACHING.md#module-5-onnx-export for:
- ONNX format benefits
- Export process walkthrough
- Verification strategies

### Acceptance Criteria
- [ ] ONNX file created successfully
- [ ] ONNX file readable by onnxruntime
- [ ] ONNX output matches PyTorch within 1e-3 tolerance
- [ ] Dynamic axes work for different batch sizes
- [ ] Benchmark shows speedup or comparable performance
- [ ] All tests in test_module_05_onnx.py pass

### Resources
- ONNX Tutorial: https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html
- ONNX Runtime Docs: https://onnxruntime.ai/docs/

---
**Labels**: tutorial, module-5, teaching, coding, testing
